en/developer_guide/block_parameter.rst

```diff
@@ -0,0 +1,115 @@
+is a part linked to the name of the parameter you will require.
+
+Backoffice work
+~~~~~~~~~~~~~~~
+
+In the Backoffice, we provide a way to store the parameter linked to the block in the block reference
+in the areas.
+
+You need to declare a service with the tag ``open_orchestra_backoffice.block_parameter.strategy``
+
+.. code-block:: yaml
+
+    tags:
+        - { name: open_orchestra_backoffice.block_parameter.strategy }
+
+The class should also implement the interface ``OpenOrchestra\Backoffice\BlockParameter\BlockParameterInterface``
+
+Frontoffice work
+~~~~~~~~~~~~~~~~
+
+In the Frontoffice, the request to display a block (in the main request) is created while you don't
+have any idea of the block type. You will have to get the block request arguments based only on the
+data stored in the ``blockParameter`` entry from the block reference.
+
+To analyse the parameters and complete the request, you need to declare a service with the tag ``open_orchestra_front.sub_query.strategy``
+
+.. code-block:: yaml
+
+    tags:
+        - { name: open_orchestra_front.sub_query.strategy }
+
+The class should also implement the interface ``OpenOrchestra\FrontBundle\SubQuery\SubQueryGeneratorInterface``
+
+Example
+-------
+
+Subquery strategies available
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
++------------------------------+------------------------------------------------------------+
+| Name                         | Action                                                     |
++==============================+============================================================+
+| ``DeviceSubQueryStrategy``   | Pass the header parameter ``x-ua-device``                  |
++------------------------------+------------------------------------------------------------+
+| ``PostDataSubQueryStrategy`` | Pass all data from the ``POST`` part                       |
++------------------------------+------------------------------------------------------------+
+| ``RequestSubQueryStrategy``  | Pass the specified request parameter (aliasId for instance |
++------------------------------+------------------------------------------------------------+
+
+New Subquery Strategy
+~~~~~~~~~~~~~~~~~~~~~
+
+Let's say that you want to translate the foo parameter from the request to give it to the block.
+
+You need to implement the three methods of the interface :
+
+ * ``getName`` : give the name of the strategy
+ * ``support`` : take the block parameter to decide if the strategy should be used
+ * ``generate`` : will generate the value
+
+In our case, we will check if the block requires a parameter named ``foo``.
+
+.. code-block:: php
+
+    public function support($blockParameter)
+    {
+        return strpos($blockParameter, 'foo') === 0;
+    }
+
+The generate method will need the ``translator`` service in order to find the translation for the parameter.
+
+.. code-block:: php
+
+    public function generate($blockParameter)
```

Comment:
Can you explain what does this method ? I assume it adds the required parameter(s) to the block call. Are they merged if several strategies are involved ? Or is this a single strategy stuff ?